File: weather-app/src/main/java/rcgonzalezf/org/weather/common/analytics/Analytics.java
package rcgonzalezf.org.weather.common.analytics;

import androidx.annotation.NonNull;

import rcgonzalezf.org.weather.WeatherApp;

public class Analytics {

  public void trackOnScreen(String screenName) {
    WeatherApp.getAnalyticsManager().notifyOnScreenLoad(screenName);
  }

  public void trackOnActionEvent(@NonNull final AnalyticsEvent event) {
    WeatherApp.getAnalyticsManager().notifyOnAction(event);
  }
}

File: weather-app/src/main/java/rcgonzalezf/org/weather/common/analytics/Analytics.kt
package rcgonzalezf.org.weather.common.analytics

import rcgonzalezf.org.weather.WeatherApp

open // for Mockito
class Analytics(private val analyticsManager: AnalyticsManager = WeatherApp.analyticsManager) {
    open fun trackOnScreen(screenName: String) {
        analyticsManager.notifyOnScreenLoad(screenName)
    }

    open fun trackOnActionEvent(event: AnalyticsEvent) {
        analyticsManager.notifyOnAction(event)
    }
}
